src/NovelRenderObject.cpp

```diff
@@ -39,13 +49,10 @@ void NovelRenderObject::configureObjectBuffers(const bool refreshBuffers) {
       bottomRight.getX(), bottomRight.getY(), 0.0f,
   };
 
-  if (refreshBuffers) {
-    glGenVertexArrays(1, &_vertexArrayObject);
-    glGenBuffers(1, &_buffer);
-  }
+  _vertexArrayObject.getActual(); //this is just here to force initialisation.
```

Comment:
Does this need to be here? It doesn't look to be used anywhere by this function...

Reply:
Yes otherwise the VAO will not exist and I figured using a `Lazy` would be easier than adding my own custom check around the VAO generation call. Also it makes teardown based on `Lazy` objects easier.

Reply:
Right, but at what point does it need to be created, because it will eventually be created in the respective `drawObject` call.

Reply:
the VAO must exist before the first buffer is created. Buffers are tied to VAOs as their "parent" so to speak, hence it is created before anything else.